### PR TITLE
Generalize PayID to support more than XRP

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Generalize PayID to support multiple networks.

### Context of Change

We need a PayIDClient which is flexible enough to deal in arbitrary networks but can still be strongly typed for use in the Xpring component. 

This PR:
- Refactors `PayIDClient` to take a string based network
- Subclasses `PayIDClient` to `XRPPayIDClient` which takes an enum based network (strong typing) and converts to a string for the super class.

This is a breaking API change but these APIs aren't finalized as of yet. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Flexible PayID!

## Test Plan

CI

### CODE REVIEW

File: src/PayID/xrp-pay-id-client.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import PayIDClient from './pay-id-client'
+import XRPLNetwork from '../Common/xrpl-network'
+
+/**
+ * Provides functionality for XRP in the PayID protocol.
+ */
+export default class XRPPayIDClient extends PayIDClient {
```
Comment (austin-king) [inline src/PayID/xrp-pay-id-client.ts:7]: great

File: src/PayID/pay-id-client.ts [hunk 2]
```diff
@@ -24,28 +22,36 @@ interface PayIDComponents {
...
-  async xrpAddressForPayID(payID: string): Promise<string> {
+  async addressForPayID(payID: string): Promise<string> {
     const payIDComponents = PayIDClient.parsePayID(payID)
 
     const client = new ApiClient()
     client.basePath = `https://${payIDComponents.host}`
 
     // Accept only the given network in response.
-    const accepts = [`application/xrpl-${this.network}+json`]
+    const accepts = [`application/${this.network}+json`]
```
Comment (hbergren) [inline src/PayID/pay-id-client.ts:54]: So `this.network` = `xrpl-mainnet` or `ach`?

**Comment by hbergren** [comment]
Also, do you know why CI is failing?

File: src/PayID/pay-id-client.ts (hunk 2 above)
Comment (keefertaylor) [reply src/PayID/pay-id-client.ts:54]: correct. We leave it as a generic string so that you can place whatever you'd like into this.

**Comment by keefertaylor** [comment]
I had a few bad names left over on a refactor that I had forgotten to push.

**Pull request merged**